### DISCUSSION TITLE
Many-to-many relation table questions - updating from Prisma 1

### DISCUSSION THREAD

**Discussion by mikestecker** [prisma/prisma#6810] [Q&A] [answered]
Can you point me in the right direction to updating this old Prisma 1 schema to Prisma 2? I'm slowly updating an old project and I have a number of user-to-user relationship tables that store other information. I don't know if it's just late in the day and my mind is fried but I'm stuck on these...

Here is my old schema:

```prisma
type UserView {
  id: Int! @id
  viewedBy: User! @relation(name: "Viewer")
  user: User! @relation(name: "UserViewed")
  count: Int!  @default(value: 1)
  isNew: Boolean! @default(value: true)
  createdAt: DateTime! @createdAt
  updatedAt: DateTime! @updatedAt
}

type FavoredUser {
  id: Int! @id
  favoredBy: User! @relation(name: "FavoredBy")
  user: User! @relation(name: "UserFavored")
  isNew: Boolean! @default(value: true)
  createdAt: DateTime! @createdAt
  updatedAt: DateTime! @updatedAt
}

type UserNote {
  id: ID! @id
  author: User! @relation(name: "NoteAuthor")
  user: User! @relation(name: "NoteForUser")
  note: String
  createdAt: DateTime! @createdAt
  updatedAt: DateTime! @updatedAt
}

type BlockedUser {
  id: Int! @id
  blockedBy: User! @relation(name: "BlockedBy")
  user: User! @relation(name: "BlockedUser")
  createdAt: DateTime! @createdAt
  updatedAt: DateTime! @updatedAt
}

type HiddenUser {
  id: Int! @id
  hiddenBy: User! @relation(name: "HiddenBy")
  user: User! @relation(name: "HiddenUser")
  createdAt: DateTime! @createdAt
  updatedAt: DateTime! @updatedAt
}
```

**Comment by mikestecker** [comment, marked as answer]
I think I got it... Only going to show one here for example of what I'm doing but I'll apply the same principle to the others:

```prisma
model User {
  [other fields]
  
  views    UserViews[] @relation("views")
  viewedBy UserViews[] @relation("userViewed")
}

model UserViews {
  id         Int      @id @default(autoincrement())
  lastViewed DateTime @default(now())

  count Int     @default(1)
  isNew Boolean @default(true)

  user         User @relation("views", fields: [userId], references: [id])
  userViewed   User @relation("userViewed", fields: [userViewedId], references: [id])
  userId       String
  userViewedId String
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
